lib/configuration/configuration_parser.js

```diff
@@ -0,0 +1,50 @@
+const Logger = require('../logger');
+const fsPromises = require('fs/promises');
+const clientConfiguration = require('./client_configuration');
+
+exports.parse = async function (configFilePath) {
+  const configFile = await tryToFind(configFilePath);
+  return configFile == null ? null : tryToParse(configFile);
+};
+
+function tryToFind (filePath) {
+  if (!filePath) {
+    return Promise.resolve(null);
+  }
+  return fsPromises.readFile(filePath, { encoding: 'utf8' })
+    .catch(err => {
+      const errorMessage = 'Finding easy logging configuration failed';
+      Logger.getInstance().error(errorMessage + '\n' + err.toString());
+      throw new Error(errorMessage);
+    });
+}
+
+function tryToParse (configurationJson) {
+  try {
+    const parsedConfiguration = JSON.parse(configurationJson);
+    validate(parsedConfiguration);
+    return new clientConfiguration.ClientConfig(
+      logLevel(parsedConfiguration),
+      logPath(parsedConfiguration)
+    );
+  } catch (e) {
+    const errorMessage = 'Parsing easy logging configuration failed';
+    Logger.getInstance().error(errorMessage + '\n' + e.toString());
+    throw new Error(errorMessage);
```

Comment:
you can pass error as a cause here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Error

Reply:
It is not working:
```
throw new Error(errorMessage, {cause: err});
```

I think probably it is not so easy in older nodejs version. So I will leave it as is for now.